lib/core/src/base64.cpp

```diff
@@ -4,135 +4,139 @@
-/**
-   base64 Encode a buffer (NUL terminated)
-   @param in      The input buffer to encode
-   @param inlen   The length of the input buffer
-   @param out     [out] The destination of the base64 encoded data
-   @param outlen  [in/out] The max size and resulting size
-   @return OK if successful
-*/
-int base64_encode( const unsigned char *in,  unsigned long inlen,
-                   unsigned char *out, unsigned long *outlen ) {
-    unsigned long i, len2, leven;
-    unsigned char *p;
+    static const unsigned char mymap[256] = {
+        255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+        255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+        255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+        255, 255, 255, 255, 255, 255, 255,  62, 255, 255, 255,  63,
+        52,  53,  54,  55,  56,  57,  58,  59,  60,  61, 255, 255,
+        255, 254, 255, 255, 255,   0,   1,   2,   3,   4,   5,   6,
+        7,   8,   9,  10,  11,  12,  13,  14,  15,  16,  17,  18,
+        19,  20,  21,  22,  23,  24,  25, 255, 255, 255, 255, 255,
+        255,  26,  27,  28,  29,  30,  31,  32,  33,  34,  35,  36,
+        37,  38,  39,  40,  41,  42,  43,  44,  45,  46,  47,  48,
+        49,  50,  51, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+        255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+        255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+        255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+        255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+        255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+        255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+        255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+        255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+        255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+        255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+        255, 255, 255, 255
+    };
 
-    /* valid output size ? */
-    len2 = 4 * ( ( inlen + 2 ) / 3 );
-    if ( *outlen < len2 + 1 ) {
-        return BASE64_BUFFER_OVERFLOW;
-    }
-    p = out;
-    leven = 3 * ( inlen / 3 );
-    for ( i = 0; i < leven; i += 3 ) {
-        *p++ = codes[( in[0] >> 2 ) & 0x3F];
-        *p++ = codes[( ( ( in[0] & 3 ) << 4 ) + ( in[1] >> 4 ) ) & 0x3F];
-        *p++ = codes[( ( ( in[1] & 0xf ) << 2 ) + ( in[2] >> 6 ) ) & 0x3F];
-        *p++ = codes[in[2] & 0x3F];
-        in += 3;
-    }
-    /* Pad it if necessary...  */
-    if ( i < inlen ) {
-        unsigned a = in[0];
-        unsigned b = ( i + 1 < inlen ) ? in[1] : 0;
+    /**
+       base64 Encode a buffer (NUL terminated)
+       @param in      The input buffer to encode
+       @param inlen   The length of the input buffer
+       @param out     [out] The destination of the base64 encoded data
+       @param outlen  [in/out] The max size and resulting size
+       @return OK if successful
+    */
```

Comment:
Let's move this to the header file.

The contents needs to be fixed to match how we document things now. See https://github.com/irods/irods/blob/c3c62ad4cfe8064bcc1475839cc79f362d0e1caa/lib/administration/user/include/irods/user_administration.hpp for examples.

Reply:
done